server.js

```diff
@@ -6287,6 +6287,92 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// MicroBadger integration.
+camp.route(/^\/microbadger\/(image-size|layers)\/([^/]+)\/([^/]+)\/?([^/]*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var type = match[1];
+  var user = match[2];
+  var repo = match[3];
+  var tag = match[4];
+  var format = match[5];
+  if (user === '_') {
+    user = 'library';
+  }
+  var path = user + '/' + repo;
+  var url = 'https://api.microbadger.com/v1/images/' + path;
+
+  var badgeData = getBadgeData(type, data);
+  if (type === 'image-size') {
+    badgeData.text[0] = getLabel('image size', data);
+  }
+
+  var options = {
+    method: 'GET',
+    uri: url,
+    headers: {
+      'Accept': 'application/json'
+    }
+  };
+  request(options, function(err, res, buffer) {
+    if (res && res.statusCode === 404) {
+      badgeData.text[1] = 'not found';
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    if (err != null || !res || res.statusCode !== 200) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    try {
+      var parsedData = JSON.parse(buffer);
+      var image;
+
+      if (tag) {
+        if (!parsedData['Versions']) {
+          badgeData.text[1] = 'not found';
+          sendBadge(format, badgeData);
+          return;
+        }
+
+        for (var i = 0; !image && i < parsedData['Versions'].length; i++) {
+          var versionJson = parsedData['Versions'][i];
+          for (var j = 0; !image && j < versionJson['Tags'].length; j++) {
+            var tagJson = versionJson['Tags'][j];
+            if (tag === tagJson.tag) {
+              image = versionJson;
+            }
+          }
+        }
+      } else {
+        image = parsedData;
+      }
+
+      if (tag && !image) {
+        badgeData.text[1] = 'not found';
+        sendBadge(format, badgeData);
+        return;
+      }
+
+      if (type == 'image-size') {
+        var size = metric(image['DownloadSize']) + "B";
```

Comment:
Could you use our `prettyBytes` helper function here?

Reply:
Thanks! I like it